src/database/remove-remote.ts

```diff
@@ -0,0 +1,126 @@
+"use strict";
+
+import { Response } from "request";
+import * as request from "request";
+import responseToError = require("../responseToError");
+import utils = require("../utils");
+import FirebaseError = require("../error");
+import logger = require("../logger");
+import api = require("../api");
+
+class RemoveRemote {
+  private instance: string;
+
+  constructor(instance: string) {
+    this.instance = instance;
+  }
+
+  public deletePath(path: string): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      const url =
+        utils.addSubdomain(api.realtimeOrigin, this.instance) + path + ".json?print=silent";
+      const reqOptions = {
+        url,
+        json: true,
+      };
+      return api.addRequestHeaders(reqOptions).then((reqOptionsWithToken) => {
+        request.del(reqOptionsWithToken, (err: any, res: Response, body: any) => {
+          if (err) {
+            return reject(
+              new FirebaseError("Unexpected error while removing data at " + path, {
+                exit: 2,
+                original: err,
+              })
+            );
+          } else if (res.statusCode >= 400) {
+            return reject(responseToError(res, body));
+          }
+          logger.debug("[database] Sucessfully removed data at " + path);
```

Comment:
I'd suggest this, just for some nice template string usage and the ability to quote your path:

```suggestion
          logger.debug(`[database] Sucessfully removed data at "${path}"`);
```

You can do this or not, I'll leave it up to you